modules/distribution/src/repository/jaggeryapps/mdm/api/user-api.jag

```diff
@@ -46,23 +52,49 @@ if (uriMatcher.match("/{context}/api/user/login/")) {
     });
 } else if (uriMatcher.match("/{context}/api/user/devices/")) {
     if (userModule.isAuthorized("/permission/device-mgt/user/devices/list")) {
-        var deviceModule = require("/modules/device.js").deviceModule;
-        var carbonUser = session.get(constants.USER_SESSION_KEY);
+        carbonUser = session.get(constants.USER_SESSION_KEY);
         result = deviceModule.listDevicesForUser(carbonUser.username);
-    }else {
+    } else {
         response.sendError(403);
     }
 } else if (uriMatcher.match("/{context}/api/users/{username}/invite")) {
     if (userModule.isAuthorized("/permission/device-mgt/admin/users/invite")) {
-        var deviceModule = require("/modules/device.js").deviceModule;
-        var elements = uriMatcher.elements();
-        var username = elements.username;
+        elements = uriMatcher.elements();
+        username = elements.username;
         userModule.inviteUser(username);
-    }else {
+    } else {
         response.sendError(403);
     }
+} else if (uriMatcher.match("/{context}/api/user/add")) {
+    if (userModule.isAuthorized("/permission/device-mgt/admin/users/add-user")) {
+        addUserFormData = request.getContent();
+
+        username = addUserFormData.username;
+        firstname = addUserFormData.firstname;
+        lastname = addUserFormData.lastname;
+        emailAddress = addUserFormData.emailAddress;
+
+        if (!addUserFormData.userRoles) {
+            userRoles = null;
+        } else {
+            userRoles = String(addUserFormData.userRoles).split(",");
+        }
+
+        try {
+            result = userModule.addUser(username, firstname, lastname, emailAddress, userRoles);
+        } catch (e) {
+            log.error("Exception in modules/user.js/addUser() : " + e.message);
+            // http status code 400 refers to - Bad request.
```

Comment:
Improve logging. Either have log.error(msg,e) or the following.
log.error(msg)
log.error(e)